service/collector/ec2instances.go

```diff
@@ -0,0 +1,241 @@
+package collector
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	clientaws "github.com/giantswarm/aws-operator/client/aws"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	// labelInstance is the metric's label key that will hold the ec2 instance ID.
+	labelInstance = "ec2instance"
+
+	// labelInstanceState is a label that will contain the instance state string
+	labelInstanceState = "state"
+
+	// labelInstanceStatus is a label that will contain the instance status check value
+	labelInstanceStatus = "status"
+
+	// subsystemEC2 will become the second part of the metric name, right after namespace.
+	subsystemEC2 = "ec2"
+)
+
+var (
+	ec2InstanceStatus = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, subsystemEC2, "status"),
+		"Gauge indicating the status of an EC2 instance. 1 = healthy, 0 = unhealthy",
+		[]string{
+			labelInstance,
+			labelAccount,
+			labelCluster,
+			labelInstallation,
+			labelOrganization,
+			labelInstanceState,
+			labelInstanceStatus,
+		},
+		nil,
+	)
+)
+
+// EC2InstancesConfig is this collector's configuration struct.
+type EC2InstancesConfig struct {
+	Helper *helper
+	Logger micrologger.Logger
+
+	InstallationName string
+}
+
+// EC2Instances is the main struct for this collector.
+type EC2Instances struct {
```

Comment:
The type suggests the filename `ec2_instances.go` to be consistent.

Reply:
Changed